src/wp-includes/class-wp-script-modules.php

```diff
@@ -363,4 +366,119 @@ private function get_src( string $id ): string {
 
 		return $src;
 	}
+
+	/**
+	 * Print data associated with Script Modules.
+	 *
+	 * The data will be embedded in the page HTML and can be read by Script Modules on page load.
+	 *
+	 * @since 6.6.0
+	 *
+	 * Data can be associated with a Script Module via the
+	 * {@see "script_module_data_{$module_id}"} filter.
+	 *
+	 * The data for a Script Module will be serialized as JSON in a script tag with an ID of the
+	 * form `wp-script-module-data-{$module_id}`.
+	 */
+	public function print_script_module_data(): void {
+		$modules = array();
+		foreach ( array_keys( $this->get_marked_for_enqueue() ) as $id ) {
+			$modules[ $id ] = true;
+		}
+		foreach ( array_keys( $this->get_import_map()['imports'] ) as $id ) {
+			$modules[ $id ] = true;
+		}
+
+		foreach ( array_keys( $modules ) as $module_id ) {
+			/**
+			 * Filters data associated with a given Script Module.
+			 *
+			 * Script Modules may require data that is required for initialization or is essential
+			 * to have immediately available on page load. These are suitable use cases for
+			 * this data.
+			 *
+			 * The dynamic portion of the hook name, `$module_id`, refers to the Script Module ID
+			 * that the data is associated with.
+			 *
+			 * This is best suited to pass essential data that must be available to the module for
+			 * initialization or immediately on page load. It does not replace the REST API or
+			 * fetching data from the client.
+			 *
+			 * @example
+			 *   add_filter(
+			 *     'script_module_data_MyScriptModuleID',
+			 *     function ( array $data ): array {
+			 *       $data['script-needs-this-data'] = 'ok';
+			 *       return $data;
+			 *     }
+			 *   );
+			 *
+			 * If the filter returns no data (an empty array), nothing will be embedded in the page.
+			 *
+			 * The data for a given Script Module, if provided, will be JSON serialized in a script
+			 * tag with an ID of the form `wp-script-module-data-{$module_id}`.
+			 *
+			 * The data can be read on the client with a pattern like this:
+			 *
+			 * @example
+			 *   const dataContainer = document.getElementById( 'wp-script-module-data-MyScriptModuleID' );
+			 *   let data = {};
+			 *   if ( dataContainer ) {
+			 *     try {
+			 *       data = JSON.parse( dataContainer.textContent );
+			 *     } catch {}
+			 *   }
+			 *   initMyScriptModuleWithData( data );
+			 *
+			 * @since 6.6.0
```

Comment:
```suggestion
			 * @since 6.7.0
```

(There might be more instances this.)